core-bundle/tests/Messenger/WebWorkerTest.php

```diff
@@ -0,0 +1,182 @@
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Messenger\Command\ConsumeMessagesCommand;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+use Symfony\Component\Messenger\RoutableMessageBus;
+use Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface;
+
+class WebWorkerTest extends TestCase
+{
+    private EventDispatcher $eventDispatcher;
+
+    private ConsumeMessagesCommand $command;
+
+    private LoggerInterface $logger;
+
+    protected function setUp(): void
+    {
+        $this->logger = new class() extends AbstractLogger {
+            private array $logs = [];
+
+            public function log($level, $message, array $context = []): void
+            {
+                $this->logs[] = $message;
+            }
+
+            public function getLogs(): array
+            {
+                return $this->logs;
+            }
+        };
+        $this->eventDispatcher = new EventDispatcher();
+        $this->createConsumeCommand();
+    }
+
+    public function testPingOnlyAboutConfiguredTransports(): void
+    {
+        $cache = $this->createMock(CacheItemPoolInterface::class);
+        $cache
+            ->expects($this->atLeastOnce()) // Tests that transport-2 is not called
+            ->method('getItem')
+            ->with('contao-web-worker-transport-1')
+        ;
+
+        $webWorker = new WebWorker(
+            $cache,
+            $this->command,
+            ['transport-1'],
+        );
+        $this->addEventsToEventDispatcher($webWorker);
+        $this->triggerRealWorkers(['transport-1', 'transport-2']);
+    }
+
+    public function testWorkerIsStoppedIfIdle(): void
+    {
+        $cache = $this->createCache(); // No real workers running
+
+        $webWorker = new WebWorker(
+            $cache,
+            $this->command,
+            ['transport-1'],
+        );
+        $this->addEventsToEventDispatcher($webWorker);
+        $this->triggerWebWorker();
+
+        // This test would run for 30 seconds if it failed. If the worker is correctly
+        // stopped, it will return immediately and log "Stopping worker.".
+        $this->assertLoggerContainsMessage('Stopping worker.');
+    }
+
+    private function createCache(array $transportsWithRunningWorkers = []): CacheItemPoolInterface
+    {
+        $cache = new ArrayAdapter();
+
+        foreach ($transportsWithRunningWorkers as $transport) {
+            $item = $cache->getItem('contao-web-worker-'.$transport);
+            $item->expiresAfter(60);
+            $cache->save($item);
+        }
+
+        return $cache;
+    }
+
+    private function triggerWebWorker(): void
+    {
+        $this->eventDispatcher->dispatch(new TerminateEvent(
+            $this->createMock(HttpKernelInterface::class),
+            new Request(),
+            new Response(),
+        ));
+    }
+
+    private function triggerRealWorkers(array $transports): void
```

Comment:
<img width="705" alt="" src="https://github.com/contao/contao/assets/1192057/fec9f6b1-087f-4ce4-8a1a-e0a176523dd0">

The given argument is always the same. Is this redundant or did you forget to add a test that provides different transports?

Reply:
No it's the same nonsense information as in the other review. It's true that it's always the same, because currently only one test uses this method. But it's still correctly residing in it's own method so it **could** be used by additional tests.